src/deluge/gui/ui/sample_marker_editor.cpp

```diff
@@ -623,6 +659,39 @@ static const uint32_t zoomUIModes[] = {UI_MODE_HOLDING_HORIZONTAL_ENCODER_BUTTON
 
 ActionResult SampleMarkerEditor::horizontalEncoderAction(int32_t offset) {
 
+	if (loopLocked && Buttons::isShiftButtonPressed()) {
+		if (offset > 0) { // turn clockwise
+			int end = getCurrentSampleHolder()->endPos;
+			int loopEnd = getCurrentMultisampleRange()->sampleHolder.loopEndPos;
+
+			if (loopEnd + loopLength < end) {
+				loopLength = loopLength * 2;
```

Comment:
is this right? This reads like it's doubling/halfing the loop length when you turn the knob instead of incrementing it which is not how I read the conversation

Reply:
Yes, I thought we might be at different viewpoints here. The intention is that you'll set a loop length related to the tempo of the sample or track then doubling or halving the loop length will keep it rhythmically intact. Being able to arbitrarily adjust the loop length with SSHIFT+TURN<> would be fine but I'd still want a want to exactly double or half it. It's a magic way to find cool and unusual looping sections within samples.